### PR TITLE
Docs: Update Format API how to guide

### LINKED ISSUE #36695: Docs: apply how to guide template to Format API guide
## What problem does this address?

PR #36694 introduces a new how to guide template, it would be good to convert the existing content to use this template. Getting a few reference examples created shows future documentation a good starting point.

[The Format API guide](https://developer.wordpress.org/block-editor/how-to-guides/format-api/) is a good first candidate to convert, as it is a contained specific task: adding a new formatting button.

## What is your proposed solution?

Update the Format API guide to a single page guide following the template from #36694.

Initially it was not clear whether we should create multiple page guides, one step per page, or everything on a single page. This confusion arose due to long tutorials and shorter how to guides, and not a clear distinction between the two. 

For how to guides, it is better for a single page, even with numerous steps because it allows the learner to see all the steps laid out on a single page. It also encourages the writer to stay focused on the single task, since a large blank page for each step might tempt the writer to fill it with lengthier explanations around concepts, while the goal of a how to guide is focused on the specific task.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Closes #36695

Update the Format API to a single page how to guide, following the new how to guide template, and adding links to the prerequisites such as plugin and JavaScript information. The goal is to focus on the format API and leave teaching more about plugins and JavaScript to other tutorials.

- Removes ES5 examples in favor of single JSX
- Update screenshots for new toolbar (WP 5.9)
- Added build instructions and troubleshooting
- Move links to references to the bottom to avoid link distractions

## How has this been tested?

Documentation changes. Check for:

- Clarity: content makes sense and accurate,
- Mistakes: check for typos or spelling issues,
- Complete: Anything missing or questions unanswered?

## Types of changes

Documentation. Condenses the [previous multi-part format API tutorial](https://developer.wordpress.org/block-editor/how-to-guides/format-api/) down to a single page.

### CODE REVIEW

File: docs/how-to-guides/format-api.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+You can also confirm by switching to HTML view (Code editor Ctrl+Shift+Alt+M) and see the text selection wrapped with `<samp>` HTML tags.
+
+Use the `className` option when registering to add your own custom class to the tag. You can use that class and custom CSS to target that element and style as you wish.
+
+### Step 4: Show the button only for specific blocks (Optional)
+
+By default, the button is rendered on every rich text toolbar (image captions, buttons, paragraphs, etc). You can render the button only on blocks of a certain type by using `wp.data.withSelect` together with `wp.compose.ifCondition`.
+
+Here is an example that onle shows the button for Paragraph blocks:
```
Comment (ryanwelcher) [inline docs/how-to-guides/format-api.md]:
```suggestion
Here is an example that only shows the button for Paragraph blocks:
```

File: docs/how-to-guides/format-api.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+
+const MyCustomButton = ( props ) => {
+	return (
+		<RichTextToolbarButton
+			icon="editor-code"
+			title="Sample output"
+			onClick={ () => {
+				props.onChange(
+					toggleFormat( props.value, {
```
Comment (ryanwelcher) [inline docs/how-to-guides/format-api.md]: Wondering if we should use destructuring for props? My guess is that is what would be seen in production code.

**Review by ryanwelcher — APPROVED** [review]
Looks good to me. Just one suggestion and a question about using destructuring - not blockers though

File: docs/how-to-guides/format-api.md (hunk 2 above)
Comment (mkaz) [reply docs/how-to-guides/format-api.md]: Switched here and the gutenberg-examples repo. 👍

**Review by ryanwelcher — APPROVED** [review]
LGTM 👍

**Pull request merged**